lib/ngx/encoding.lua

```diff
@@ -0,0 +1,92 @@
+-- Copyright (C) by Yichun Zhang (agentzh)
```

Comment:
I don't like the `ngx.encoding` module name. Can we make it `ngx.base64`?

Also, we need some docs for this module :)

Reply:
Changed. But I was thinking about `ngx.encoding` so that we may be able to add more variants of encodings in the future without introducing a new module.

Reply:
@dndx I prefer smaller Lua modules since it can save memory footprint and GC overhead by skip loading stuff the user does not need.